node/src/ledger/ledger_manager.rs

```diff
@@ -0,0 +1,627 @@
+        let (sender, receiver) = mpsc::unbounded_channel();
+        let runtime = thread::spawn(move || {
+            let mut ledger_ctx = ledger_ctx;
+            let mut receiver = receiver;
+            loop {
+                match receiver.blocking_recv() {
+                    Some(LedgerRequestWithChan { request, responder }) => responder
+                        .send(request.handle(&mut ledger_ctx))
+                        .unwrap_or(()),
+                    None => {
+                        break;
+                    }
+                }
+            }
+            ledger_ctx
+        });
+        LedgerManager { sender, runtime }
+    }
+
+    fn call(&self, request: LedgerRequest) -> Result<LedgerResponse, String> {
+        let (responder, receiver) = channel();
+        self.sender
+            .send(LedgerRequestWithChan { request, responder })
+            .unwrap();
+        result_join(receiver.recv())
+    }
+
+    pub async fn wait_for_stop(
+        self,
+    ) -> Result<LedgerCtx, Box<dyn std::any::Any + std::marker::Send + 'static>> {
+        self.runtime.join()
+    }
+
+    pub fn insert_genesis_ledger(&self, mask: Mask) {
+        self.call(LedgerRequest::InsertGenesisLedger { mask })
+            .unwrap();
+    }
+
+    pub fn get_mask(&self, ledger_hash: &LedgerHash) -> Option<(Mask, bool)> {
+        match self.call(LedgerRequest::GetMask {
+            ledger_hash: ledger_hash.clone(),
+        }) {
+            Ok(LedgerResponse::LedgerMask(mask)) => mask,
+            _ => panic!("get_mask failed"),
+        }
+    }
+    pub fn producers_with_delegates(
+        &self,
+        ledger_hash: &LedgerHash,
+        filter: fn(&CompressedPubKey) -> bool,
+    ) -> Option<BTreeMap<AccountPublicKey, Vec<(ledger::AccountIndex, AccountPublicKey, u64)>>>
+    {
+        match self.call(LedgerRequest::GetProducersWithDelegates {
+            ledger_hash: ledger_hash.clone(),
+            filter,
+        }) {
+            Ok(LedgerResponse::ProducersWithDelegatesMap(map)) => map,
+            _ => panic!("producers_with_delegates failed"),
+        }
+    }
+}
+
+fn result_join<T>(r: Result<Result<T, String>, RecvError>) -> Result<T, String> {
+    match r {
+        Ok(Ok(v)) => Ok(v),
+        Ok(Err(e)) => Err(e),
+        Err(e) => Err(e.to_string()),
+    }
+}
+
+impl redux::TimeService for LedgerManager {}
+
+impl redux::Service for LedgerManager {}
+
+impl TransitionFrontierSyncLedgerSnarkedService for LedgerManager {
+    fn compute_snarked_ledger_hashes(
+        &self,
+        snarked_ledger_hash: &LedgerHash,
+    ) -> Result<(), String> {
+        self.call(LedgerRequest::ComputeSnarkedLedgerHashes {
+            snarked_ledger_hash: snarked_ledger_hash.clone(),
+        })
+        .and_then(|res| {
+            if let LedgerResponse::Success = res {
+                Ok(())
+            } else {
+                Err("LedgerManager::compute_snarked_ledger_hashes: unexpected response".to_string())
```

Comment:
lets include the response in all these error results (should never happen, but it if does it will be easier to figure out what is going on):

```suggestion
                Err(format!("LedgerManager::compute_snarked_ledger_hashes: unexpected response: {res:?}"))
```

Reply:
Done.